lib/elixir/test/elixir/kernel_test.exs

```diff
@@ -456,6 +456,24 @@ defmodule KernelTest do
       assert case_in(-3, -1..-3) == true
     end
 
+    def map_dot(map) when map.field, do: true
+    def map_dot(_other), do: false
+
+    test "map dot guard" do
+      refute map_dot(:foo)
+      refute map_dot(%{})
+      refute map_dot(%{field: false})
+      assert map_dot(%{field: true})
+
+      message = ~r"cannot invoke remote function map.field/0 inside guards"
+
+      assert_raise CompileError, message, fn ->
+        defmodule MapDot do
+          def map_dot(map) when map.field(), do: true
+        end
+      end
+    end
+
```

Comment:
Would this support 

```
def map_dot(map) when map.child.grandchild.field(), do: true
```
?

Reply:
Without the parenthesis in `.field()` it will work.

Reply:
Exciting!! Nice one :heart_eyes: